translated/tech/20190530-Benefits-of-Using-Microservice-Architecture-in-Go.md

```diff
@@ -0,0 +1,98 @@
+# 在 Go 中使用微服务架构的好处
+
+## 前言
+我们已经讨论“微服务架构”很长一段时间了。它是软件架构中最新的热门话题。那么什么是微服务呢?我们为什么要使用它?为什么要在 Golang 中使用微服务架构?它有哪些优点？
+
+本文中，我将会探讨一些相关的问题。废话不多说，让我们开始吧。
+
+## 什么是微服务？
+微服务是一种软件开发技术，属于 SOA（面向服务的架构）的一种形式。它的作用是，将应用程序构建为许多松耦合的服务的集合。在这种架构中，服务的编码通常是细粒度的，服务的协议更轻量。目前还没有对微服务的准确定义，但它有一些显著的特征：自动化部署、业务功能、去中心化的数据管理和智能端点。
+（译者注：对于这个定义有兴趣的同学，不妨去看看 Martin 大神[关于微服务的文章](https://martinfowler.com/articles/microservices.html)
+
+## 我们为什么使用微服务？
+这种架构帮助我们将整个应用程序拆分成各个部分或小模块，从而更容易理解、开发和测试。它帮助我们将服务看作是独立的，但又清楚地指定了在应用程序中使用它。更进一步地，它有助于保持项目架构的一致性(最初设计的架构和实际开发完成的架构差别不大)。它还可以通过建立不同的独立团队来进行服务的部署和扩展，从而各团队能够并行地开发。在这个架构中重构代码更容易。它也支持连续交付和部署流程（CI/CD)。
+
+## 为什么使用 go 构建微服务？
+在回答这个问题之前。首先，我说一下 Golang 的优势。虽然 Golang 是一门新的语言，但是与其他语言相比，它有很多优势。其次，用 Golang 编写的程序更加健壮。它们能够承受程序使用运行的服务构建的繁重负载。Golang 更适合多处理器系统和 web 应用程序。此外，它与 GitHub 集成，便于管理非集中的代码包。微服务架构的用处大部分体现在当程序需要伸缩（scalable)时。如果有一种语言可以完全符合标准,那么它就是Golang。原因是它继承自 C-family 编程语言，用 Golang 编写的组件更容易结合组件编码在其他语言中。
```

Comment:
it does is because of its inheritance from C-family programming languages, the components written in golang are easier to combine with components coded in other languages that reside in the same family.
reside in the same family 这个修饰词应该加上